### ISSUE TITLE
FEATURE - Allow for metadata fields to be included in staging models

### ISSUE THREAD

**Issue by fivetran-joemarkiewicz** [fivetran/dbt_stripe_source#20]
**Are you a Fivetran customer?** 

Original request identified by @sachams 

**Is your feature request related to a problem? Please describe.**
 
No, this is to add the `metadata` fields to staging models.

**Describe the solution you'd like**
 
Ideally a customer would be able to add the metadata field and possible parse the field out as well.

**Current PR Work**
PR #18

**Comment by fivetran-joemarkiewicz** [comment]
@sachams I wanted to loop back up here to let you know our team has spent most of this week working on dbt 20 compatibility updates. We are going to focus on upgrading this package as well as our other packages and then we will be able to tackle this request once we have completed all our upgrades.

**Comment by mnahkies** [comment]
Just wanted to add that I'm interested in using the stripe connector, but would also require the metadata fields to perform the kind of analysis / reconciliation that I'm planning.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks for commenting @mnahkies! Would you be able to specify which metadata field for which object you would be interested in? I am still trying to scope out how our team will approach this task and it would be helpful to get more insight into which metadata fields you are most interested in and for which table(s).

Thanks!

**Comment by mnahkies** [comment]
We are interested in the `subscription` metadata specifically. 

Given that the contents are custom key->value pairs, I would be satisfied with either a target column `metadata JSON` / `TEXT` column containing a serialized JSON object.

Alternatively the proposal of putting the content into columns like `metadata_my_key` sounded fine too.

Since writing my comment, I have realized that we should be able to fulfill our use case by using the `subscription_id` to then reconstruct the metadata from our own records that are loaded from another source, so perhaps it isn't as important for us as I initially thought.

**Comment by sachams** [comment]
Hi @fivetran-joemarkiewicz how's the work going? 

By the way, a colleague recently set up the Mixpanel DBT package and was able to use the [event properties pivot](https://github.com/fivetran/dbt_mixpanel#pivoting-out-event-properties) to extract custom fields from JSON into columns. Could this approach be used for Stripe fields?

**Comment by fivetran-joemarkiewicz** [comment]
Hi @sachams I was able to get some more information from @mnahkies and the approach you pointed out for Mixpanel was exactly the route I was considering going! 

Our team has still been pretty busy the last few weeks and we have not been able to tackle this feature at the moment. With this in mind I will add this ticket to our upcoming sprint (starting tomorrow) and will keep you both updated for when a working branch is ready to be tested out.

Thanks as well @sachams for the PR #18 as this will be extremely helpful!

**Comment by fivetran-joemarkiewicz** [comment]
@mnahkies and @sachams I have been able to add the functionality for the `charge`, `invoice`, `payment_intent`, `payment_method`, `payout`, `plan`, `refund`, and `subscription` staging models to pivot out defined metadata fields within PR #28. Huge thanks to @sachams for the bulk of the PR work 🎉 

It would be great if you would be able to read through the updates and give the new functionality a try and let me know if it all looks good on your end. You can do so by using the below config in your `packages.yml` and be sure to update your variables configuration based off the documentation I added in the PR for the metadata pivot.

```yml
packages:
    - git: https://github.com/fivetran/dbt_stripe_source.git
      revision: customer/sachams-metadata-add
      warn-unpinned: false
```

**Comment by fivetran-joemarkiewicz** [comment]
Hi All, 

Just wanted to provide an update that the PR #28 has been merged and was included in the `v0.4.3` release of the stripe_source package! This should be live on the dbt hub at the top of the hour. Thanks again for all of your help with integrating this into the package. 

As this has been included in the latest release I will close this issue. Please feel free to open a new issue if you have any other issues or feature requests. Thanks again!
